go/vt/schemadiff/table_test.go

```diff
@@ -705,3 +706,104 @@ func TestCreateTableDiff(t *testing.T) {
 		})
 	}
 }
+
+func TestValidate(t *testing.T) {
+	tt := []struct {
+		name      string
+		from      string
+		alter     string
+		expectErr error
+	}{
+		{
+			name:  "add column",
+			from:  "create table t (id int primary key)",
+			alter: "alter table t add column i int",
+		},
+		{
+			name:  "add key",
+			from:  "create table t (id int primary key, i int)",
+			alter: "alter table t add key i_idx(i)",
+		},
+		{
+			name:  "add column and key",
+			from:  "create table t (id int primary key)",
+			alter: "alter table t add column i int, add key i_idx(i)",
+		},
+		{
+			name:      "add key, missing column",
+			from:      "create table t (id int primary key, i int)",
+			alter:     "alter table t add key j_idx(j)",
+			expectErr: ErrInvalidColumnInKey,
+		},
+		{
+			name:      "add key, missing column 2",
+			from:      "create table t (id int primary key, i int)",
+			alter:     "alter table t add key j_idx(j, i)",
+			expectErr: ErrInvalidColumnInKey,
+		},
+		{
+			name:  "drop column, ok",
+			from:  "create table t (id int primary key, i int, i2 int, key i_idx(i))",
+			alter: "alter table t drop column i2",
+		},
+		{
+			name:      "drop column, break key",
+			from:      "create table t (id int primary key, i int, key i_idx(i))",
+			alter:     "alter table t drop column i",
+			expectErr: ErrInvalidColumnInKey,
+		},
```

Comment:
Right now this makes sense as a safeguard until we improve the drop column logic to also drop references from the index to mimic MySQL behavior.

Reply:
If we choose to go that path then I'll update this PR; no need to postpone that to a future PR.